### ISSUE TITLE
Bidding logic always interpreted as v0

### ISSUE THREAD

**Issue by typerat** [android/privacy-sandbox-samples#106]
Hi 👋 

We're using bidding logic v3 and adding the header `x_fledge_buyer_bidding_logic_version 3` following the [mock server specification](https://github.com/android/privacy-sandbox-samples/blob/4b16a4cb4a842657fd135568e9578d4750916570/Fledge/FledgeServerSpec/mock-server-v3-bidding-logic.json#L30). This has been working for some months now, until recently the API stopped respecting that header. The logs always show  `adservices.fledge: Got buyer bidding logic version 0`.

Any idea why?

**Comment by galarragas** [comment]
Would you be able to check the value of this flag `fledge_ad_selection_bidding_logic_js_version` using 

```
device_config get adservices fledge_ad_selection_bidding_logic_js_version
```

If the version is left to the default value of 2 [we are not reading](https://cs.android.com/android/platform/superproject/main/+/main:packages/modules/AdServices/adservices/service-core/java/com/android/adservices/service/adselection/AdBidGeneratorImpl.java;l=239) the response version header. It is possible that the server flag controlling the version has been changed so you are now seeing the different in behavior .

If this is the case you can enable V3 support by using

```
device_config put adservices fledge_ad_selection_bidding_logic_js_version 3
```

**Comment by typerat** [comment]
Hi @galarragas, thanks for your response!
Running `adb shell device_config get adservices fledge_ad_selection_bidding_logic_js_version` returns `null`.

> If the version is left to the default value of 2 we are not reading the response version header.

Why isn't the default behavior the same as in the documentation? This is confusing... I'd expect the header to be read _always_. How is this supposed to work on user devices where we can't set flags?

**Comment by typerat** [comment]
@galarragas I understand that B&A is your priority. However, this has been working in the past. We actively made the switch from v0 to v3 some months ago. We did lots of debugging and got it working eventually by adding the header. We ran successful auctions with both versions. Why is it broken now?

**Comment by galarragas** [comment]
> Why isn't the default behavior the same as in the documentation? This is confusing... I'd expect the header to be read always.

I agree with you, we have features that are documented and not ramped up yet so there is a disconnect. 

>  How is this supposed to work on user devices where we can't set flags?

On user device we are currently only supporting the old logic. Once the feature will be ramped up your server will receive an header `X_FLEDGE_1_VERSION ` declaring what is the version of JS logic supported by the client. 

We are going to improve the documentation, sorry for the confusion.
